Library/Homebrew/build.rb

```diff
@@ -100,6 +100,15 @@ def install
         ENV.prepend "LDFLAGS", "-L#{dep.opt_lib}" if dep.opt_lib.directory?
         ENV.prepend "CPPFLAGS", "-I#{dep.opt_include}" if dep.opt_include.directory?
       end
+
+      if OS.linux?
+        formula_deps.each do |dep|
+          ENV.append_path "LD_RUN_PATH", dep.opt_lib
+        end
+        unless ENV["LD_RUN_PATH"].empty?
+          ENV.prepend "LDFLAGS", "-Wl,-rpath,#{ENV["LD_RUN_PATH"]}"
```

Comment:
Minor thing: /home/linuxbrew/.linuxbrew/lib may show up twice in the rpath after doing this.

As a test:

```
% brew install -s mksh monit
% patchelf --print-rpath /home/linuxbrew/.linuxbrew/bin/mksh
/home/linuxbrew/.linuxbrew/lib
% patchelf --print-rpath /home/linuxbrew/.linuxbrew/bin/monit
/home/linuxbrew/.linuxbrew/lib:/home/linuxbrew/.linuxbrew/lib:/home/linuxbrew/.linuxbrew/opt/zlib/lib:/home/linuxbrew/.linuxbrew/opt/openssl/lib
```